src/ray/core_worker/store_provider/memory_store/memory_store.h

```diff
@@ -51,7 +51,10 @@ class CoreWorkerMemoryStore {
       std::shared_ptr<ReferenceCounter> counter = nullptr,
       std::shared_ptr<raylet::RayletClient> raylet_client = nullptr,
       std::function<Status()> check_signals = nullptr,
-      std::function<void(const RayObject &)> unhandled_exception_handler = nullptr);
+      std::function<void(const RayObject &)> unhandled_exception_handler = nullptr,
+      std::function<std::shared_ptr<RayObject>(const RayObject &object,
+                                               const ObjectID &object_id)>
+          object_allocator = nullptr);
```

Comment:
should we have inline comments on the guarantee of this object_allocator API provide? i.e. when will it the buffer being freed? 
note: if we change the Buffer to JvmManagedBuffer, we might need a different abstraction.

Reply:
JvmManagedBuffer is not general enough if we're going to port more language frontends like Golang, Rust.